server/src/test/java/org/elasticsearch/index/replication/IndexLevelReplicationTests.java

```diff
@@ -368,6 +372,44 @@ public void testSeqNoCollision() throws Exception {
         }
     }
 
+    /**
+     * This test ensures the consistency between primary and replica with late and out of order delivery on the replica.
+     * An index operation on the primary is followed by a delete operation. The delete operation is delivered first
+     * and processed on the replica but the index is delayed with an interval that is even longer the gc deletes cycle.
+     * This makes sure that that replica still remembers the delete operation and correctly ignores the stale index operation.
+     */
+    public void testLateDeliveryAfterGCTriggeredOnReplica() throws Exception {
+        try (ReplicationGroup shards = createGroup(1)) {
+            shards.startAll();
+            final IndexShard primary = shards.getPrimary();
+            final IndexShard replica = shards.getReplicas().get(0);
+            final TimeValue gcInterval = TimeValue.timeValueMillis(scaledRandomIntBetween(1, 1000));
```

Comment:
I think we can just set this to something very small (10ms?) and also set ThreadPool#ESTIMATED_TIME_INTERVAL_SETTING to 0?

Reply:
done